### PR TITLE
add installation config of CR, provide examples

### PR DESCRIPTION
**Pull request by erezo9**
#### Summary
This adds the option to enable the user the installation of the mattermost CR. instead of looking for examples at the operator
also provided link where the user can view the full spec to take example

### CODE REVIEW

**Comment by erezo9** [comment]
I was also able to produce multiple mattermost CR with one single values.yaml ( because the operator lets you manage multiple isntance)
if this is the preffered idea, i can add a commit to it

```

  specs:
    mattermost1:
      namespace: mattermost1
      name: mattermost-example
      spec:
        version: 7.2.0
        replicas: 1
        ingress:
          enabled: true
          host: example.mattermost-example.com
        database:
          operatorManaged:                            # Operator managed database allows configuration of replicas, storage size and resource requirements. If the `size` is specified, it will override those values.
            replicas: 1
            resources:
              requests:
                cpu: 250m
                memory: 512Mi
            storageSize: 50Gi
            type: mysql
        fileStore:
          operatorManaged:                            # Operator managed file store allows configuration of replicas, storage size and resource requirements. If the `size` is specified, it will override those values.
            replicas: 1
            resources:
              requests:
                cpu: 150m
                memory: 512Mi
            storageSize: 50Gi
    mattermost2:
      namespace: mattermost2
      name: mattermost-example
      spec:
        version: 7.2.0
        replicas: 1
        ingress:
          enabled: true
          host: example.mattermost-example.com
        database:
          operatorManaged:                            # Operator managed database allows configuration of replicas, storage size and resource requirements. If the `size` is specified, it will override those values.
            replicas: 1
            resources:
              requests:
                cpu: 250m
                memory: 512Mi
            storageSize: 50Gi
            type: mysql
        fileStore:
          operatorManaged:                            # Operator managed file store allows configuration of replicas, storage size and resource requirements. If the `size` is specified, it will override those values.
            replicas: 1
            resources:
              requests:
                cpu: 150m
                memory: 512Mi
            storageSize: 50Gi

```


```
{{- if and .Values.mattermostCR.enabled }}
{{- $dot := . }}
{{- range $configuration, $Mydata := $.Values.mattermostCR.specs }}
---
apiVersion: installation.mattermost.com/v1beta1
kind: Mattermost
metadata:
  creationTimestamp: null
  namespace: {{ $Mydata.namespace }}
  name: {{ $Mydata.name }}
  labels:
    helm.sh/chart: {{ $dot.Chart.Name }}-{{ $dot.Chart.Version | replace "+" "_" }}
    app.kubernetes.io/name: {{ template "mattermost-operator.name" $ }}
    app.kubernetes.io/managed-by: {{ $dot.Release.Service }}
    app.kubernetes.io/instance: {{ $dot.Release.Name }}
spec:
{{ toYaml $Mydata.spec | indent 2}}
{{- end }}
{{- end }}
```

**Comment by stafot** [comment]
@Szymongib PTAL on the PR and on the [comment](https://github.com/mattermost/mattermost-helm/pull/370#issuecomment-1264645848)

**Review by Szymongib — APPROVED** [review]
Thanks, LGTM!

I think we can go with a single CR for now.

File: charts/mattermost-operator/values.yaml [hunk 1]
```diff
@@ -2,6 +2,38 @@
 # This is a YAML-formatted file.
 # Declare variables to be passed into your templates.
 
+# The values beloew are for the mattermost custom resource
```
Comment (phoinixgrr) [inline charts/mattermost-operator/values.yaml:5]:
```suggestion
# The values below are for mattermost Custom Resource
```

File: tests/e2e-kind.sh [hunk 2]
```diff
@@ -28,7 +28,7 @@ if [[ "$cluster_name" != "${CLUSTER_NAME}" ]]; then
 fi
 
 docker run --rm -u "$(id -u):$(id -g)" --interactive --network host \
-    -entrypoint '/bin/sh' \
+    --entrypoint '/bin/sh' \
```
Comment (phoinixgrr) [inline tests/e2e-kind.sh:31]: Nice catch 👍

**Review by phoinixgrr — COMMENTED** [review]
Works great here: 

```sh
> kubectl get mattermosts.installation.mattermost.com                                                                                           
NAME                 STATE         IMAGE   VERSION   ENDPOINT
mattermost-example   reconciling
```

Helm template: 
```
[...]
---
# Source: mattermost-operator/templates/mattermost-operator/mattermost.yaml
apiVersion: installation.mattermost.com/v1beta1
kind: Mattermost
metadata:
  creationTimestamp: null
  name: mattermost-example
  labels:
    helm.sh/chart: mattermost-operator-0.3.2
    app.kubernetes.io/name: mattermost-operator
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/instance: release-name
spec:
  database:
    operatorManaged:
      replicas: 1
      resources:
        requests:
          cpu: 250m
          memory: 512Mi
      storageSize: 50Gi
      type: mysql
  fileStore:
    operatorManaged:
      replicas: 1
      resources:
        requests:
          cpu: 150m
          memory: 512Mi
      storageSize: 50Gi
  ingress:
    enabled: false
    host: example2.mattermost-example.com
  replicas: 1
  version: 7.2.0
```

```
helm lint .  --strict                                                                                                                         ⎈ kind-hakis/skata
==> Linting .
[INFO] Chart.yaml: icon is recommended

1 chart(s) linted, 0 chart(s) failed
```

Left a small editorial comment

**Comment by phoinixgrr** [comment]
Tests [fail](https://github.com/mattermost/mattermost-helm/actions/runs/3196619209/jobs/5277826910) with: 
```
Error: Error installing charts: Error adding repo: incubator=https://charts.helm.sh/incubator: Error waiting for process: exit status 1
Error installing charts: Error adding repo: incubator=https://charts.helm.sh/incubator: Error waiting for process: exit status 1
``` 

This seems to happen because `ct` runs with non root privileges, and requires to write in the filesystem under root `/.config` folder to update helm repos. 

@stafot @stylianosrigas  @angeloskyratzakos @spirosoik 
Is this something known? Seems to not be related with the changes introduced in this PR. Tests should fail for everyone

**Comment by phoinixgrr** [comment]
Raised https://github.com/mattermost/mattermost-helm/pull/372 to fix failing tests

@erezo9 When the above is merged, feel free to update your fork, and rebase from master to get the fix.

**Comment by erezo9** [comment]
@phoinixgrr I have rebased and pushed based on your fix
you should be able to run the test now

**Comment by phoinixgrr** [comment]
> @phoinixgrr I have rebased and pushed based on your fix
> you should be able to run the test now

Thank you
Tests registered and passing :rocket:

**Review by phoinixgrr — APPROVED** [review]

**Pull request merged**